src/Microsoft.ServiceFabric.Actors/Runtime/MigrationActorStateProvider.cs

```diff
@@ -0,0 +1,445 @@
+        }
+
+        /// <inheritdoc/>
+        public Task<IReplicator> OpenAsync(ReplicaOpenMode openMode, IStatefulServicePartition partition, CancellationToken cancellationToken)
+        {
+            this.servicePartition = partition;
+            return ((IStateProviderReplica)this.rcStateProvider).OpenAsync(openMode, partition, cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public Task ChangeRoleAsync(ReplicaRole newRole, CancellationToken cancellationToken)
+        {
+            return ((IStateProviderReplica)this.rcStateProvider).ChangeRoleAsync(newRole, cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public Task CloseAsync(CancellationToken cancellationToken)
+        {
+            return ((IStateProviderReplica)this.rcStateProvider).CloseAsync(cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public void Abort()
+        {
+            ((IStateProviderReplica)this.rcStateProvider).Abort();
+        }
+
+        /// <inheritdoc/>
+        public Task BackupAsync(Func<BackupInfo, CancellationToken, Task<bool>> backupCallback)
+        {
+            return ((IStateProviderReplica)this.rcStateProvider).BackupAsync(backupCallback);
+        }
+
+        /// <inheritdoc/>
+        public Task BackupAsync(BackupOption option, TimeSpan timeout, CancellationToken cancellationToken, Func<BackupInfo, CancellationToken, Task<bool>> backupCallback)
+        {
+            return ((IStateProviderReplica)this.rcStateProvider).BackupAsync(option, timeout, cancellationToken, backupCallback);
+        }
+
+        /// <inheritdoc/>
+        public Task RestoreAsync(string backupFolderPath)
+        {
+            return ((IStateProviderReplica)this.rcStateProvider).RestoreAsync(backupFolderPath);
+        }
+
+        /// <inheritdoc/>
+        public Task RestoreAsync(string backupFolderPath, RestorePolicy restorePolicy, CancellationToken cancellationToken)
+        {
+            return ((IStateProviderReplica)this.rcStateProvider).RestoreAsync(backupFolderPath, restorePolicy, cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public Task OnSnapshotAsync(TimeSpan currentLogicalTime)
+        {
+            return ((VolatileLogicalTimeManager.ISnapshotHandler)this.rcStateProvider).OnSnapshotAsync(currentLogicalTime);
+        }
+
+        /// <summary>
+        /// Modifies data from KVS store into a suitable format for RC. Saves the modified data in RC store.
+        /// </summary>
+        /// <param name="keyValuePairs">
+        /// Data from KVS store that needs to be modified and saved in RC
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Cancellation token
+        /// </param>
+        /// <returns>A <see cref="Task"/> representing the result of the asynchronous operation.</returns>
+        public async Task SaveStatetoRCAsync(List<KeyValuePair<string, byte[]>> keyValuePairs, CancellationToken cancellationToken)
+        {
```

Comment:
We need to log all the keys migrated in NOISE level and number of actor states, reminders, etc migrated in INFO level

Reply:
Done